app/src/main/java/com/example/bookwormadventuresdeluxe2/Notification.java

```diff
@@ -0,0 +1,48 @@
+package com.example.bookwormadventuresdeluxe2;
+
+import android.widget.ImageView;
+
+import com.example.bookwormadventuresdeluxe2.Utilities.Status;
+
+public class Notification
+{
+    private Book book;
+    private String message;
+
+    public Notification(Book book, String message)
```

Comment:
Could we just add notification to the book class? At first I was thinking this could look like inheritance but I guess notification does not have isa relationship with Book.

Reply:
I could make notifications a book with the description as the message because notifications don't need a description.
But I think having a separate class is better especially if later we want to have a notification factory or something.

Reply:
or make it a subclass! and just call super for book details? I haven't tried thinking of any conflicts. this is great though

Reply:
Yeah I kinda agree with Kate on this one; it doesn't pass the is-a test but it definitely has a has-a relationship.

Reply:
Made it so Notification extends book and added createNotification method to book

Reply:
I'm sorry that this is coming up after you've made the changes but what Richmond and I were trying to explain is that the relationship between Notification and Book is **not** an is-a relationship, therefore inheritance is not appropriate. Instead, since Notification has the qualities of a Book, it is a has-a relationship. This is reflected in the UML from part 2. I think the appropriate solution here would have been to remove the methods that are already defined in Book (lines 18-46 in your original commit) and only call book.getThing() when necessary instead of redefining the methods. I think my original idea to put Notification inside of Book was wrong but the goal here is to get rid of the repeated code

Reply:
reverted Notification extends Book, removed duplicate calls in Notification, kept createNotification() in Book